cl/users/email_backend.py

```diff
@@ -0,0 +1,317 @@
+from threading import local
+
+from django.conf import settings
+from django.core.mail import (
+    EmailMessage,
+    EmailMultiAlternatives,
+    get_connection,
+)
+from django.core.mail.backends.base import BaseEmailBackend
+
+from cl.lib.command_utils import logger
+from cl.users.email_handlers import (
+    convert_list_to_str,
+    get_email_body,
+    store_message,
+)
+from cl.users.models import OBJECT_TYPES, BackoffEvent, EmailFlag
+
+
+class ConnectionHandler:
+    """This is an email backend connection handler, based on
+    django-post-office/connections.py
+    """
```

Comment:
Sure I've reviewed this in detail and concluded that is not necessary to use the connection handler because we are using only one base email backed (django-ses), the reason that django-post-office uses this it's because it allows you to choose a different base backend when sending a message, so they open and close connections inside the send messages loop, so in order to avoid creating multiple connections for each message they use the connection handler to reuse the same connection during the same backend call.

So I deleted that connection handler and instead of opening the connection inside the send messages loop, we open and close the connection outside the loop, so we use the same connection for the same backend call.

Reply:
Great news. That was a big piece of seemingly-complicated boilerplate. Glad to be rid of it.